addons/common/fnc_randomizeFacewear.sqf

```diff
@@ -0,0 +1,44 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_randomizeFacewear
+
+Description:
+    Add config defined weighted random facewear to unit.
+    Uses same config as headgearList[].
+
+Parameters:
+    _unit - unit <OBJECT>
+
+Returns:
+    true on success, false on error <BOOLEAN>
+
+Examples:
+    (begin example)
+        [unit] call CBA_fnc_randomizeFacewear
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+SCRIPT(randomizeFacewear);
+
+params [["_unit", objNull]];
+
+if (isNull _unit) exitWith {
+    TRACE_1("unit is null",_unit);
+    false
+};
+
+if (!local _unit || {!(_unit getVariable ["BIS_enableRandomization", true])}) exitWith {true};
+
+private _allowedFacewear = getArray (configFile >> "CfgVehicles" >> typeOf _unit >> "CBA_allowedFacewear");
+
+if (_allowedFacewear isEqualTo []) exitWith {true};
+
+private _facewear = selectRandomWeighted _allowedFacewear;
+
+if (_facewear == "") then {
```

Comment:
This is changed from BWA version, which uses `"None"`. I did this to keep it same as `headgearList[]`.

Reply:
This is good. BI seems to have changed their system, it used NONE dummy class in CfgGlasses when I wrote the original.

Reply:
Thought more about this.

BI still has this `CfgGlasses/None` class. While public scope, it clearly is not intended to be worn. It even has a bug where it displays the DLC icon in inventory (yet doesn't trigger the DLC ads spam).

I'd suggest this just to be rigorous:
```suggestion
if (toLower _facewear in ["", "none"]) then {
```